validate/validate.go

```diff
@@ -5,12 +5,103 @@ import (
 	"fmt"
 	"github.com/hashicorp/hcl/v2"
 	"io/ioutil"
+	"regexp"
 )
 
-// WordWithDotHyphenUnder80Pattern is a regexp pattern that matches string that contain alphanumerics, underscores, periods,
+type RegexpValidator struct {
+	pattern       string
+	errorTemplate string
+	regex         *regexp.Regexp
+}
+
+// Check validates provided string with a regexp based on the pattern and returns optional error.
+func (r *RegexpValidator) Check(value string, valueType interface{}) error {
+	if r.regex == nil {
```

Comment:
Having this logic here means that if the regexp fails to compile, the user will get a validation error with that, instead of an internal server error.

Since these are static expressions, I would prefer a panic instead of returning an error. For more dynamic errors that could happen, I would change the Validate method to also return an error besides the validation errors, but don't think it's needed for this case.

Reply:
Yes, I think you are right. I've been back and forth with this and I think I'm gonna change it - had same doubts as you described.